node/network/gossip-support/src/lib.rs

```diff
@@ -209,6 +220,12 @@ where
 						"New session detected",
 					);
 					self.last_session_index = Some(session_index);
+
+					// Update the authority status metric on every new session.
+					match util::Validator::new(leaf, self.keystore.clone(), ctx.sender()).await {
+						Ok(_) => self.metrics.on_is_authority(),
+						Err(_) => self.metrics.on_is_not_authority(),
```

Comment:
we should probably ignore runtime errors

Reply:
definitely